tests/acceptance/stepDefinitions/sharingContext.js

```diff
@@ -444,22 +445,49 @@ When('the user disables all the custom permissions of collaborator {string} for
     .disableAllCustomPermissions(collaborator)
 })
 
-Then('custom permission/permissions {string} should be set for user {string} for file/folder {string} on the webUI', function (permissions, user, resource) {
-  return client.page
+const assertSharePermissions = async function (currentSharePermissions, permissions = undefined) {
+  let expectedPermissionArray
+  if (permissions !== undefined) {
+    expectedPermissionArray = await client.page.FilesPageElement.sharingDialog().getArrayFromPermissionString(permissions)
+  }
+  const COLLABORATOR_PERMISSION_ARRAY = ['share', 'update', 'create', 'delete']
+  for (let i = 0; i < COLLABORATOR_PERMISSION_ARRAY.length; i++) {
+    const permissionName = COLLABORATOR_PERMISSION_ARRAY[i]
+    if (permissions !== undefined) {
+      // check all the required permissions are set
+      if (expectedPermissionArray.includes(permissionName)) {
+        assert.strictEqual(currentSharePermissions[permissionName], true, `Permission ${permissionName} is not set`)
+      } else {
+        // check unexpected permissions are not set or absent from the array
+        assert.ok(!currentSharePermissions[permissionName], `Permission ${permissionName} is set`)
+      }
+    } else {
+      // check all the permissions are not set or absent from the array
+      assert.ok(!currentSharePermissions[permissionName], `Permission ${permissionName} is set`)
+    }
+  }
+}
+
+Then('custom permission/permissions {string} should be set for user {string} for file/folder {string} on the webUI', async function (permissions, user, resource) {
```

Comment:
This line is too long, can you split this line

Reply:
done